.travis.yml

```diff
@@ -11,6 +11,7 @@ before_script:
 
 script:
   - mkdir -p build/logs
+  - vendor/bin/psalm.phar
```

Comment:
Instead of running Psalm with every build, it would be nice to create a separate job for it.
You can take inspiration from brick/math: https://github.com/brick/math/blob/master/.travis.yml

Reply:
I'm not clear on what the benefit of the separate job would be at this stage. Psalm only seems to take 1.15 seconds to run, so the overheads of setting up a separate job probably wouldn't be justified by the potential savings of running it in parallel with other things. I'm also not very familiar with Travis.

Reply:
You have a point here, though it's more of a matter of proper job separation, so that when a job fails, you know that it's because of a failed test and not because of a Psalm issue. This also gives the ability to use Psalm as one of the "smoke tests" that prevent other tests from running if it doesn't pass. Let's merge it this way for now, I'll come back to this later.